parlai/core/metrics.py

```diff
@@ -10,6 +10,9 @@
 between processes.
 """
 
+__ALL__METRICS__ = ['correct', 'rouge-1', 'rouge-2', 'rouge-L', 'bleu' 'accuracy', 'f1', 'loss', 'ppl']
```

Comment:
`__` prefixes are only for hidden variables that you don't want to expose. It's not unreasonable to have these variables with `ALL_METRICS` and `DEFAULT_METRICS`.

loss/ppl shouldn't be in this list; they are agent-level metrics.

ALL_METRICS should be 'DEFAULT_METRICS + ['rouge-1', ...]`

Reply:
Thanks! I didn't know that.